apps/builder/app/canvas/instance-hovering.ts

```diff
@@ -44,27 +45,41 @@ export const subscribeInstanceHovering = () => {
       hoveredInstanceSelectorStore.set(undefined);
       hoveredInstanceOutlineStore.set(undefined);
     }, 100);
+
+    // Fixes the bug, that new hover occures during timeout
+    const unsubscribe = hoveredInstanceSelectorStore.listen(() => {
+      clearTimeout(mouseOutTimeoutId);
+      unsubscribe();
+    });
   };
 
   const eventOptions = { passive: true };
   window.addEventListener("mouseover", handleMouseOver, eventOptions);
   window.addEventListener("mouseout", handleMouseOut, eventOptions);
 
-  // debounce is used to avoid laggy hover because of iframe message delay
-  const updateHoveredRect = (element: Element) => {
-    const instanceId = element.getAttribute(idAttribute) ?? undefined;
-    if (instanceId === undefined) {
+  const updateHoveredRect = (
+    elements: Element[],
+    instanceSelector: Readonly<InstanceSelector>
+  ) => {
+    if (elements.length === 0) {
       return;
     }
+
+    if (instanceSelector.length === 0) {
```

Comment:
The utility protects from such case by returning undefined. No need to check here.

Reply:
Until I see type 
type NonEmptyArray<T> = [T, ...T[]]
I would check length, as its some internal implementation detail of selector not exposed somehow